src/lib/AgreementStructs.sol

```diff
@@ -21,8 +21,12 @@ struct Agreement {
     bytes32 termsHash;
     /// @dev Required amount to join or merkle root of (address,amount).
     uint256 criteria;
+    /// @dev Total balance deposited in the agreement.
+    uint256 balance;
     /// @dev Number of finalizations confirmations.
     uint256 finalizations;
+    /// @dev Signal if agreement is disputed.
+    bool disputed;
```

Comment:
Are _disputed_ and _not disputed_ the only states of an Agreement? If there are more possible states, maybe use an `enum` here instead of `bool`?